project-management/Github.md

```diff
@@ -0,0 +1,77 @@
+# Managing public (or private) projects using GitHub
+
+We use a common set of labels across projects in GitHub. It is
+possible to precisely control this set of labels using
+the [settings GitHub app][github-app-settings] (needs to be enabled
+for your org). Use the [`settings.yml`](../.github/settings.yml) from
+this repository as a starting point.
+
+[github-app-settings]: https://github.com/probot/settings
+
+## Labels
+
+We use labels to classify tasks by priority, from P0 to P4:
+
+* **P0 (blocker)** Do not do anything else, this is a blocker.
+* **P1 (critical)** Must have.
+* **P2 (major)** Should have.
+* **P3 (minor)** Could have.
+* **P4 (trivial)** Someday/maybe. Consider closing as "wontfix".
+
+Labels are also useful to classify issues by type or by component. Use
+colours to distinguish whether a label designates a type, a component
+or something else.
+
+## Project leads
+
+Each project must have a project lead (maintainer). By convention, the
+PL is whoever appears at the top of [`CODEOWNERS`][github-codeowners]
+file, e.g.:
+
+```
+* @mboes
+```
+
+This has the effect that @mboes will automatically be tagged as
+a reviewer for all incoming PR's, unless later lines in the same file
+override that.
+
+[github-codeowners]: https://help.github.com/articles/about-codeowners/
+
+## Milestones
+
+We gate tickets with milestones. Milestones always have an associated
+due date.
```

Comment:
I'm more or less guessing what you are getting at here. But this needs to be more specific. Like « Each tickets must have an associated milestone. Milestone are created <some way>. Milestones must have an associated due date <explain how to attach a new date>. »

We should also explain what are the expectations for a due date.

(is there a time semantics inside github? So that we can easily see things which are past their planned date).

Reply:
> (is there a time semantics inside github? So that we can easily see things which are past their planned date).

Yes.

> We should also explain what are the expectations for a due date.

Could you clarify? I don't want to be too normative here. Projects have their own constraints when it comes to timing and tempo. The only thing that I think is important is that *if* you have milestones defined, then that only makes sense if there's a date associated to it (GitHub lets the date be optional).

Reply:
Well, there are a number of things which you can read in a date: the milestone _must_ be delivered at date T, otherwise something has fail, OR based on the amount of work that we believe will be needed, we estimate that the milestone will be delivered around this time, OR at due time, whether the milestone is reached or not, everything in the milestone should be reassigned to new milestones, for the sake of better planning. I probably forget some.

While I agree that being too normative is probably counter productive, it is my impression that misunderstandings in the meaning of dates is a frequent cause of problems. To avoid this, I believe it would be better if the document gives a baseline understanding of what we usually mean by a due date.

Reply:
This is all completely project specific.